File: Backends/contabil-java/src/main/java/br/com/talison/contabil/domain/dto/UserDto.java
package br.com.talison.contabil.domain.dto;

import lombok.Getter;
import lombok.Setter;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.Date;

@Getter
@Setter
public class UserDto implements Serializable {

    private String id;

    @NotNull
    @NotBlank
    private String name;

    @NotNull
    @NotBlank
    private String role;

    private Date createdAt;


}

File: Backends/contabil-go/src/domain/userDTO.go
package domain

import (
	_ "gorm.io/gorm"
)

type UserDTO struct {
	ID        string `json:"id" gorm:"type:varchar(255);primary_key;"`
	Name      string `json:"name" gorm:"type:varchar(255);not null;unique;idx_user"`
	Role      string `json:"role" gorm:"type:varchar(255);not null;idx_user"`
	CreatedAt string `json:"createdAt" gorm:"type:varchar(255);not null;idx_user"`
	UpdatedAt string `json:"updatedAt" gorm:"type:varchar(255);not null;idx_user"`
}
